ev3dev/core.py

```diff
@@ -605,9 +616,20 @@ def time_sp(self):
     @time_sp.setter
     def time_sp(self, value):
         self.set_attr_int('time_sp', value)
+# ~autogen
 
+    def wait_for_running(self):
+        while True:
+            if 'running' in self.state:
+                return
+            sleep(0.01)
```

Comment:
> You might get some pushback on the use of lambda to create the anonymous methods started and stopped but I think it's elegant.

`started` and `stopped` in my example are not anonymous methods, they are just `bool`s. If `wait_for()` exits because the condition was met, it returns `True`. When the timeout has been hit, it returns `False`.